tests/test_core.py

```diff
@@ -736,27 +736,35 @@ async def test_coro():
     assert len(call_count) == 3
 
 
-async def test_async_add_job_pending_tasks_coro(hass: HomeAssistant) -> None:
+def test_add_job_pending_tasks_coro(hass: HomeAssistant) -> None:
     """Add a coro to pending tasks."""
-    call_count = []
 
     async def test_coro():
         """Test Coro."""
-        call_count.append("call")
+        pass
 
     for _ in range(2):
         hass.add_job(test_coro())
 
-    async def wait_finish_callback():
-        """Wait until all stuff is scheduled."""
-        await asyncio.sleep(0)
-        await asyncio.sleep(0)
+    # Ensure add_job does not run immediately
+    assert len(hass._tasks) == 0
 
-    await wait_finish_callback()
+
+async def test_async_add_job_pending_tasks_coro(hass: HomeAssistant) -> None:
```

Comment:
This test was named to test the async api, but it ran the sync api in the event loop